lib/jsdom/living/helpers/form-controls.js

```diff
@@ -60,6 +64,50 @@ exports.normalizeToCRLF = string => {
     .replace(/^\n/, "\r\n");
 };
 
+exports.isLabelable = node => {
+  // labelable logic defined at: https://html.spec.whatwg.org/multipage/forms.html#category-label
+  if (node.nodeType !== NODE_TYPE.ELEMENT_NODE) {
+    return false;
+  }
+
+  switch (node.tagName) {
+    case "BUTTON":
+    case "METER":
+    case "OUTPUT":
+    case "PROGRESS":
+    case "SELECT":
+    case "TEXTAREA":
+      return true;
+
+    case "INPUT":
+      return node.type !== "hidden";
+  }
+
+  return false;
+};
+
+exports.getLabelsForLabelable = labelable => {
+  if (!exports.isLabelable(labelable)) {
+    return null;
+  }
+  if (!labelable._labels) {
```

Comment:
Can you declare `this._labels = undefined` in the constructors of all classes that use this function?

Reply:
Initialized `this._labels = null` everywhere.